---
title: " "
author: " "
date: " "
bibliography: dataanalysis_ref.bib 
geometry: "margin=2.5cm" 
linestretch: 1.5
fontsize: 12pt
output:
  pdf_document: 
    #toc: true
    #toc_depth: 2
    number_sections: true
    
---
\pagenumbering{arabic}
\begin{titlepage}
\centering
	{\Large Ruprecht-Karls-Universität Heidelberg\\
		Fakultät für Biowissenschaften\\
		Bachelorstudiengang Molekulare Biotechnologie\\}

	{\vspace{\stretch{3}}}


		{\Huge Cancer Hallmark and Metabolic Pathways differ over Cancer types and in Prostate Adenocarcinoma patients}

		

	

	\vspace{\stretch{4}}
	{\Large Data Science Project SoSe 2022}

	\vspace{\stretch{0.25}}
	
{\Large
	 `r format(Sys.time(),'%d %B %Y')`
}
	\vspace{\stretch{0.75}}

	{\Large
		Fabian Strobel, Lottida Phondeth, Laura Lange, Carla Welz  
}

	\vspace{\stretch{1}}

\end{titlepage}

\newpage
\tableofcontents

\newpage
# Introduction
## Hallmarks of cancer
## Prostate adenocarcinoma
... LOTTI



Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R. [@RN1]

![Caption for the picture.](descriptive_analysis/Disribution of age at initial pathologic diagnosis.png)




***
# Methods
## Initial raw data
During our project, we used four given data sets. The first was an R-object consisting of a list of gene sets for cancer hallmarks. Second, a pan-cancer RNA-seq gene expression data frame for 9,741 patients of 33 various cancer types based on data generated by the “The Cancer Genome Atlas” Research Network: https://www.cancer.gov/tcga. In addition, there was an R-object containing 37 clinical annotations regarding the RNA-seq patients. And fourth, for a focused analysis of PRAD, an R-object with RNA-seq gene expression data of matched tumor tissue and normal tissue of 52 PRAD patients was used.
To get a broader view about the cancer hallmarks and metabolic activities, additional gene sets were chosen from the Molecular Signatures Database (MSigDB) (Liberzon et al., 2011; Subramanian et al., 2005) after literature review. Trying to get a large overlap with the genes from the RNA-seq, 509 additional gene sets were used, resulting in a total number of 555 gene sets used during the study. These include 50 hallmark gene sets (Liberzon et al., 2015), 186 curated gene sets from KEGG pathway database (Kanehisa, 2019; Kanehisa et al., 2021; Kanehisa and Goto, 2000), 189 oncogenic signature gene sets (Liberzon et al., 2011; Subramanian et al., 2005) and the 84 largest ontology gene sets (Ashburner et al., 2000; GOC, 2021; Köhler et al., 2020) as of June 2022.


## Preprocessing
The RNA-seq data came in a log2(TPM) format which served as normalization technique. 
![Preprocessing.](report_figures/figure10.jpg)




## Analysis of gene sets
To get a sense of the meaningfulness of the used gene sets in the study, their overlap with the genes of the RNAseq was investigated using Venn diagrams. 

Before analyzing the RNA-seq data, a comparison of the gene sets was performed. Therefore, it was necessary to convert all gene names in the same format. The gene symbols in the pathways of the given gene sets were converted into Ensemble gene IDs using the *EnsDb.Hsapiens.v79 package*. Regarding the additional gene sets, the genes were also imported as Ensembl gene IDs. Next, all gene sets were combined into one list. To take a closer look at the gene sets, the similarity between them was investigated. As a metric which compares how many genes are shared between the different gene sets, the Jaccard Indices
$$
J(A, B) = \frac{|A \cap B|}{|A \cup B|} \> with \\ 0 \leq J(A, B) \leq 1
$$
were computed for each combination. These result in a value between zero and one which is the ratio between intersection and union of the two respective pathways (Levandowsky and Winter, 1971). After computing the Jaccard indices for every combination a heatmap was created which also clustered the gene sets.




## Pan-cancer analysis

### descriptive analsis --> CARLA


### linear regression




## Focused analysis: Prostate adenocarcinome
### descriptive analysis CARLA
As with the pan-cancer RNAseq data frame, PCA and UMAP were applied on a combined data frame of tumor and normal samples from RNAseq.
### Volcano Plots LAURA

### Gene ontology enrichment analysis

### GSEA/GSVA


### PRAD pathway activity matrix --> HEATMAP

After GSVA PCA and UMAP were applied on the pathway activity matrix. Furthermore, the pathway activity of the differentially expressed genes was visualized in the figure. 





***
# Results

## Preprocessing



![Preprocessing results.](report_figures/figure20.jpg)

## Gene set analysis



![Gene set analysis.](report_figures/figure30.jpg)

![Descriptive analysis. CARLA NEUE PLOTS](report_figures/figure40.jpg)

## Pan-cancer analysis


![Pan-cancer GSVA and UMAP.](report_figures/figure50.jpg)

![Correlation between cancer types after GSVA.](report_figures/figure60.jpg)

![GSVA 30 topVar pws.](report_figures/figure70.jpg)


![Volcano plots.](report_figures/figure90.jpg)

![GOEA.](report_figures/figure100.jpg)

![PRAD GSVA UMAP.](report_figures/figure110.jpg)


***
# Discussion
## kjhtgrfedws
## uztrwwret  

***
# Refrences
  