---
title: "Challenge1"
output: html_document
date: '2022-06-07'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
Aufgabe:

Schreibt eine Funktion, die folgende Inputs nimmt:

    1.) List of different dataframes (e.g. all LUAD patients, all BRCA, …)
    2.) List of different Genesets
    3.) Type of Test (Wilcoxon, t-test, …)

Die Funktion nimmt zuerst den ersten dataframe aus 1.) und außerdem nimmt sich die Funktion das erste Geneset aus 2.). Dann schaut es sich die Gene aus dem Geneset an und extrahiert die jeweiligen GeneExpression values aus dem ersten dataframe 1.) für Patient 1. Zusätzlich schaut es sich alle GeneExpression values von den Genen an, die nicht im Geneset vorhanden sind! Basierend auf den zwei Gene Expression Vectoren wird ein Test durchgeführt (Wilcoxon/oder ein anderer). Ihr bekommt einen p-value. Diesen Workflow macht ihr für jedes Geneset in 2.). Am Ende bekommt Ihr einen p-value Vektor für Patient1 in dataframe 1.) mit der Länge eurer Genesets. Das wird für alle Patienten wiederholt! Ihr bekommt für dataframe 1 aus 1.) einen neuen dataframe mit samples x genesets (p-value-matrix). Die Funktion wiederholt diesen Workflow für jeden dataframe aus 1.).Zusatz: p-value correction?!Output: list of p-value dataframes for each tumor type!



Testdaten erstellen:
```{r}
set.seed(100)

# create example data

df1 <- data.frame(matrix(rnorm(50*50, 0, 2), 50, 50), row.names = 1:50)
df2 <- data.frame(matrix(rnorm(50*50, 0, 1.5), 50, 50), row.names = 1:50)

df.list = list("LUAD" = df1, "BRCA" = df2)

pathway1 <- c(1, 4, 6, 34, 27, 14)
pathway2 <- c(5, 8, 9, 19, 5, 19, 20, 50)

pw.list = list("pathway1" = pathway1, "pathway2"=pathway2)
```


P-Value Matrix der Testdaden erstellen:
```{r}
pvalue_matrix_list<-list()
for(i in 1:length(df.list)){
single_tumortype <- data.frame(df.list[i])
   pvalue_matrix <- c()
  for(p in 1:length(pw.list)){
    include_list<-data.frame(pw.list[p])
    single_tumortype_geneset <- subset(single_tumortype,rownames(single_tumortype) %in% include_list[,1] )
    single_tumortype_wo_geneset <- subset(single_tumortype,!rownames(single_tumortype) %in% include_list[,1] )
    
    pvalue_temp <-c()
   
    for(h in 1:ncol(single_tumortype)){
             pvalue_temp[[h]] <-wilcox.test(single_tumortype_geneset[,h], single_tumortype_wo_geneset[,h])$p.value 
    }
    pvalue_matrix <- rbind(pvalue_matrix,pvalue_temp)
  }
   pvalue_matrix_list[[i]] <-  pvalue_matrix
}
names(pvalue_matrix_list) <- names(df.list)

```


P-Value Matrix der Testdaden mit apply erstellen(funktioniert nicht):
```{r}
pvalue_matrix_list2<-list()
for(i in 1:length(df.list)){
single_tumortype <- data.frame(df.list[i])
   pvalue_matrix <- c()
  for(p in 1:length(pw.list)){
    include_list<-data.frame(pw.list[p])
    #single_tumortype_geneset <- subset(single_tumortype,rownames(single_tumortype) %in% include_list[,1] )
    #single_tumortype_wo_geneset <- subset(single_tumortype,!rownames(single_tumortype) %in% include_list[,1] )
    
    pvalue_temp <-c()
   
    pvalue_temp <-apply(single_tumortype, 2, function(x) wilcox.test(
      x[subset(single_tumortype,rownames(single_tumortype) %in% include_list[,1])],
      x[subset(single_tumortype,!rownames(single_tumortype) %in% include_list[,1])])$p.value)
    
    pvalue_matrix <- rbind(pvalue_matrix,pvalue_temp)
  }
   pvalue_matrix_list2[[i]] <-  pvalue_matrix
}
names(pvalue_matrix_list) <- names(df.list)

```


Pancancer Daten laden:
```{r}
setwd('..')

tcga_pancancer <- readRDS("data/tcga_pancancer.rds")
pathways <- readRDS("data/total_pathways_ensID.rds")
```


P-Value Matrix der Pancancer Daten erstellen:
```{r}

single_tumortype <- data.frame(tcga_pancancer)
   pvalue_matrix <- c()
   #Schleife über alle Pathways
  for(p in 1:length(pathways)){
    include_list<-data.frame(pathways[p])
    single_tumortype_geneset <- subset(single_tumortype,rownames(single_tumortype) %in% include_list[,1] )
    single_tumortype_wo_geneset <- subset(single_tumortype,!rownames(single_tumortype) %in% include_list[,1] )
    
    pvalue_temp <-c()
   
    #Schleife über alle Patienten
    for(h in 1:ncol(single_tumortype)){
             pvalue_temp[[h]] <-wilcox.test(single_tumortype_geneset[,h], single_tumortype_wo_geneset[,h])$p.value 
    }
    pvalue_matrix <- rbind(pvalue_matrix,pvalue_temp)
  }
```


***
```{r}
# Computing the variance
v <- apply(tcga_pancancer, 1, var) 

# Getting genes with highest variance
topVar <- tcga_pancancer[v > quantile(v, 0.9),]
```

```{r}
topVar_genes <- rownames(topVar)

intersection <- lapply(pathways, function(pw){
  length(intersect(pw, topVar_genes))/length(union(pw, topVar_genes))
})

inter <- t(as.data.frame(intersection))

topInt <- which(inter > quantile(inter, 0.9),)

topInt_pws <- pathways[topInt]
```

```{r}
x <- list(RNAseq_topVar = topVar_genes, Pathways = unique(unlist(topInt_pws, use.names = F)))

ggVennDiagram::ggVennDiagram(x) +
  ggtitle("Venn diagram of PRAD RNAseq genes and pathway genes") +
  theme(plot.title = element_text(hjust = 0.5)) +
  ggplot2::scale_fill_gradient(low="lightsalmon",high = "lightblue2")
```

```{r}


single_tumortype <- data.frame(topVar)
   pvalue_matrix <- c()
   #Schleife über alle Pathways
  for(p in 1:length(topInt_pws)){
    print(names(topInt_pws[p]))
    include_list<-data.frame(topInt_pws[p])
    single_tumortype_geneset <- subset(single_tumortype,rownames(single_tumortype) %in% include_list[,1] )
    single_tumortype_wo_geneset <- subset(single_tumortype,!rownames(single_tumortype) %in% include_list[,1] )
    
    pvalue_temp <-c()
   
    #Schleife über alle Patienten
    for(h in 1:ncol(single_tumortype)){
             pvalue_temp[[h]] <-wilcox.test(single_tumortype_geneset[,h], single_tumortype_wo_geneset[,h])$p.value
             if (h %% 1000 == 0){
               x <- 
               print(round((h/ncol(single_tumortype)),1))
             }
               
    }
    pvalue_matrix <- rbind(pvalue_matrix,pvalue_temp)
  }
```

